packages/services/src/features/localStorage.service.ts

```diff
@@ -22,20 +22,65 @@ export const saveToLocalStorage = (data: any, parameters: { storageKey: string }
  */
 export const mergeToLocalStorage = (
   data: any,
-  parameters: { storageKey: string; objectKeyPath: string; objectValuePath?: string },
+  parameters: { storageKey: string; objectKeyPath?: string; objectValuePath?: string },
 ) => {
   try {
-    const { storageKey, objectKeyPath } = parameters;
+    const { storageKey } = parameters;
     const existingData = localStorage.getItem(storageKey);
-    const key = getValueByPath(data, objectKeyPath);
     const value = parameters.objectValuePath ? getValueByPath(data, parameters.objectValuePath) : data;
 
     if (existingData) {
       const parsedData = JSON.parse(existingData);
-      localStorage.setItem(storageKey, JSON.stringify({ ...parsedData, [key]: value }));
+      if (parameters.objectKeyPath) {
+        const key = getValueByPath(data, parameters.objectKeyPath);
+
+        localStorage.setItem(storageKey, JSON.stringify({ ...parsedData, [key]: value }));
+      } else {
+        localStorage.setItem(storageKey, JSON.stringify({ ...parsedData, ...value }));
+      }
     } else {
-      localStorage.setItem(storageKey, JSON.stringify({ [key]: value }));
+      if (parameters.objectKeyPath) {
+        const key = getValueByPath(data, parameters.objectKeyPath);
+
+        localStorage.setItem(storageKey, JSON.stringify({ [key]: value }));
+      } else {
+        localStorage.setItem(storageKey, JSON.stringify({ ...value }));
+      }
+    }
+  } catch (error: any) {
+    console.error(error);
+    throw new Error(error.message);
+  }
+};
```

Comment:
I noticed you've made significant changes to the `mergeToLocalStorage` function, but the unit tests for it haven't been updated. Could you update the unit tests to ensure that this function works as expected?

Reply:
Good point! I will update it